src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -293,7 +293,8 @@ def _do_upgrade(self):
                 })
                 return
             self.upgrade_state.target_id = target_id
-            self.upgrade_state.target_version = target_version
+            # extract the version portion of 'ceph version {version} ({sha1})'
+            self.upgrade_state.target_version = target_version.split(' ')[2]
```

Comment:
this only works, if `target_version` is set by the newly upgraded MGR. If the `self.upgrade_state.target_version` was set by the old MGR, `self.upgrade_state.target_version` will be deserialized as the full string. 

I fear that we might need to cope with `self.upgrade_state.target_version` being set already .

Reply:
added a compat kludge